### PR TITLE
[agent] Add deprecation warnings when running the process checks on the process agent

### PR DESCRIPTION
**Pull request by Stephanie0829**
### What does this PR do?
- Add deprecation warnings when running the process checks on the process agent, and advise on switch to core agent (for linux only).
- As process checks will be moving to core agent regardless, for deprecation purposes this log warning will be emitted even if NPM is enabled.
  - This PR adds minor refactoring to allow this behavior. 

### Motivation
https://datadoghq.atlassian.net/browse/PROCS-4206

### Describe how to test/QA your changes
1. Local unit testing by running the `agent_linux_test.go` tests (primarily `TestProcessAgentComponentOnLinux`),  in a linux environment. Can set breakpoints to ensure log message is being output for conditional branches.
2. Run process agent to ensure new log message is being output.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/[agent]-Add-deprecation-warnings-when-running-the-process-checks-on-the-process-agent-9a3eb8622e86e39e.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
```
Comment (wiyu) [inline releasenotes/notes/[agent]-Add-deprecation-warnings-when-running-the-process-checks-on-the-process-agent-9a3eb8622e86e39e.yaml:7]: [minor] this comment could be removed

File: releasenotes/notes/[agent]-Add-deprecation-warnings-when-running-the-process-checks-on-the-process-agent-9a3eb8622e86e39e.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+---
+enhancements:
+  - |
+    Add deprecation warnings when running process checks on the process agent in linux.
+other:
+  - |
+    Change prepares for the deprecation of Process/Container Collection in the Process Agent, occurring in a future release.
```
Comment (wiyu) [inline releasenotes/notes/[agent]-Add-deprecation-warnings-when-running-the-process-checks-on-the-process-agent-9a3eb8622e86e39e.yaml:14]: [minor] These actually go out into our [release notes](https://github.com/DataDog/datadog-agent/releases/tag/7.55.0). You can just combine them into a single section, `other` seems like a good spot, so the change makes more sense for an external reader.

File: comp/process/agent/agent_linux.go [hunk 3]
```diff
@@ -26,28 +35,35 @@ func Enabled(config config.Component, checkComponents []types.CheckComponent, lo
...
-			break
+		}
+		if slices.Contains(processCheckNames, check.Object().Name()) && check.Object().IsEnabled() {
+			processEnabled = true
 		}
 	}
 
 	switch flavor.GetFlavor() {
 	case flavor.ProcessAgent:
+
```
Comment (wiyu) [inline comp/process/agent/agent_linux.go:50]: [style] no need for this newline

File: releasenotes/notes/[agent]-Add-deprecation-warnings-9a3eb8622e86e39e.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+other:
+  - |
+    Add deprecation warnings when running process checks on the process agent in linux.
```
Comment (buraizu) [inline releasenotes/notes/[agent]-Add-deprecation-warnings-9a3eb8622e86e39e.yaml:4]:
```suggestion
    Add deprecation warnings when running process checks on the process Agent in Linux.
```

File: releasenotes/notes/[agent]-Add-deprecation-warnings-9a3eb8622e86e39e.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+    Change prepares for the deprecation of Process/Container Collection in the Process Agent, occurring in a future release.
```
Comment (buraizu) [inline releasenotes/notes/[agent]-Add-deprecation-warnings-9a3eb8622e86e39e.yaml:5]:
```suggestion
    This change prepares for the deprecation of process and container collection in the Process Agent, occurring in a future release.
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with some minor edits requested in the release note

**Review by daniel-taf — APPROVED** [review]

**Review by wiyu — APPROVED** [review]

**Pull request merged**